cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesGroupDao.java

```diff
@@ -27,70 +37,141 @@ public TimeSeriesGroupDao(DSLContext dsl)
+	private List<TimeSeriesGroup> getTimeSeriesGroupsWhere(Condition whereCond)
+	{
+		List<TimeSeriesGroup> retval = new ArrayList<>();
+		AV_TS_CAT_GRP catGrp = AV_TS_CAT_GRP.AV_TS_CAT_GRP;
+		AV_TS_GRP_ASSGN grpAssgn = AV_TS_GRP_ASSGN.AV_TS_GRP_ASSGN;
 
-		Select select = step;
+		final RecordMapper<Record, Pair<TimeSeriesGroup, AssignedTimeSeries>> mapper = record15 -> {
+			TimeSeriesGroup group = buildTimeSeriesGroup(record15);
+			AssignedTimeSeries loc = buildAssignedTimeSeries(record15);
 
-		if ( officeId != null && !officeId.isEmpty())
+			return new Pair<>(group, loc);
+		};
+
+		SelectOnConditionStep<Record16<String, String, String, String, String, String, String, String, String, String, String, String, BigDecimal, BigDecimal, String, String>>
+				selectOn = dsl.select(
+				catGrp.CAT_DB_OFFICE_ID, catGrp.TS_CATEGORY_ID, catGrp.TS_CATEGORY_DESC, catGrp.GRP_DB_OFFICE_ID,
+				catGrp.TS_GROUP_ID, catGrp.TS_GROUP_DESC, catGrp.SHARED_TS_ALIAS_ID, catGrp.SHARED_REF_TS_ID,
+				grpAssgn.CATEGORY_ID, grpAssgn.DB_OFFICE_ID, grpAssgn.GROUP_ID, grpAssgn.TS_ID,
+				grpAssgn.TS_CODE,grpAssgn.ATTRIBUTE,
+				grpAssgn.ALIAS_ID, grpAssgn.REF_TS_ID)
+				.from(catGrp)
+				.leftOuterJoin(grpAssgn)
+				.on(
+				catGrp.TS_CATEGORY_ID.eq(grpAssgn.CATEGORY_ID)
+						.and(catGrp.TS_GROUP_ID.eq(grpAssgn.GROUP_ID))
+						.and(catGrp.GRP_DB_OFFICE_ID.eq(grpAssgn.DB_OFFICE_ID)));
+
+
+		SelectOrderByStep<Record16<String, String, String, String, String, String, String, String, String, String, String, String, BigDecimal, BigDecimal, String, String>> select = selectOn;
+		if(whereCond != null)
 		{
-			select = step.where(table.GRP_DB_OFFICE_ID.eq(officeId));
+			SelectConditionStep<Record16<String, String, String, String, String, String, String, String, String, String, String, String, BigDecimal, BigDecimal, String, String>> selectWhere = selectOn.where(
+					whereCond);
+			select = selectWhere;
 		}
 
-		return select
-				//  .orderBy(AV_TS_CAT_GRP.AV_TS_CAT_GRP.TS_GROUP_ATTRIBUTE)
-				.fetch().into(TimeSeriesGroup.class);
+		List<Pair<TimeSeriesGroup, AssignedTimeSeries>> assignments = select
+				.orderBy(grpAssgn.ATTRIBUTE).fetch(mapper);
+
+		Map<TimeSeriesGroup, List<AssignedTimeSeries>> map = new LinkedHashMap<>();
+		for(Pair<TimeSeriesGroup, AssignedTimeSeries> pair : assignments){
+			List<AssignedTimeSeries> list = map.computeIfAbsent(pair.component1(), k -> new ArrayList<>());
+			AssignedTimeSeries assignedTimeSeries = pair.component2();
+			if(assignedTimeSeries != null)
+			{
+				list.add(assignedTimeSeries);
+			}
+		}
+
+		for(final Map.Entry<TimeSeriesGroup, List<AssignedTimeSeries>> entry : map.entrySet())
+		{
+			List<AssignedTimeSeries> assigned = entry.getValue();
+
+			retval.add(new TimeSeriesGroup(entry.getKey(), assigned));
+		}
+		return retval;
 	}
 
-	public List<TimeSeriesGroup> getTimeSeriesGroups(String officeId, String categoryId, String groupId)
+	private AssignedTimeSeries buildAssignedTimeSeries(Record record15)
```

Comment:
This variable should probably have a better name than record15.

Reply:
Agreed.